src/plugin.ts

```diff
@@ -4,154 +4,123 @@ import pixelMatch from 'pixelmatch'
 import { PNG } from 'pngjs'
 import sanitize from 'sanitize-filename'
 import { serializeError, type ErrorObject } from 'serialize-error'
-
 import { createFolder } from './utils/fs'
 import { adjustCanvas, parseImage } from './utils/image'
 import { logger } from './logger'
 
-let CYPRESS_SCREENSHOT_DIR = 'cypress/screenshots' // TODO why ?!
-
 export type UpdateSnapshotArgs = {
   screenshotName: string
-  specRelativePath: string
-  specFolder: string
-  screenshotsFolder: string
-  snapshotBaseDirectory: string
-}
-
-function getSpecTrimmedPath(relativePath: string, integrationFolder: string): string {
-  integrationFolder = integrationFolder ?? path.join('cypress', 'e2e')
-  return relativePath.replace(integrationFolder, '')
-}
-
-/** Update the base snapshot .png by copying the generated snapshot to the base snapshot directory.
- * The target path is constructed from parts at runtime in node to be OS independent.  */
-const updateSnapshot = async (args: UpdateSnapshotArgs): Promise<boolean> => {
-  const specTrimmedPath = getSpecTrimmedPath(args.specRelativePath, args.specFolder)
-  const toDir = args.snapshotBaseDirectory ?? path.join(process.cwd(), 'cypress', 'snapshots', 'base')
-  const snapshotActualDirectory = args.screenshotsFolder ?? path.join('cypress', 'screenshots')
-  const destDir = path.join(toDir, specTrimmedPath)
-
-  const fromPath = path.join(snapshotActualDirectory, specTrimmedPath, `${args.screenshotName}.png`)
-  const destFile = path.join(destDir, `${args.screenshotName}.png`)
-
-  await createFolder(destDir)
-  await fs.copyFile(fromPath, destFile)
-  logger.debug(`Updated base snapshot '${args.screenshotName}' at ${destFile}`)
-  return true
+  specName: string
+  screenshotAbsolutePath: string
+  baseDirectory: string
 }
 
 export type CompareSnapshotsPluginArgs = {
```

Comment:
I left the types here for now, as I tried not to focus on too many different things :D